upgrade versions: - nn 97.0.0; - maps 10.5.0-rc.1;

RingerJK opened the pull request:
<!--
---------- CHECKLIST ----------
1. Add related labels (`bug`, `feature`, `new API(s)`, `SEMVER-MAJOR`, `needs-backporting`, etc.).
1. Adda a changelog entry under `Unreleased` tag or a `skip changelog` label if not applicable.
1. Update progress status on the project board.
1. Request a review from the team, if not a draft.
1. Add targeted milestone, when applicable.
1. Create ticket tracking addition of public documentation pages entry, when applicable.
-->

LukasPaczos commented on libnavui-maps/src/test/java/com/mapbox/navigation/ui/maps/camera/data/MapboxNavigationViewportDataSourceTest.kt:
```diff
@@ -125,7 +125,7 @@ class MapboxNavigationViewportDataSourceTest {
     @Before
     fun setup() {
-        mockkStatic(DirectionsRoute::toNavigationRoute)
+        mockkStatic("com.mapbox.navigation.base.route.NavigationRouteEx")
```
Does the function reference not work?

RingerJK replied:
it's not working anymore, when a function is overloaded the compiler cannot resolve which methods must be used

LukasPaczos commented:
Where's the Kotlin stdlib update coming from? We might need to avoid it for now until we setup a strategy across the SDKs.

RingerJK commented:
> Where's the Kotlin stdlib update coming from? We might need to avoid it for now until we setup a strategy across the SDKs.

kotlin is not updated in this PR. Initially, I added an update of the kotlin version, but later separate into another PR https://github.com/mapbox/mapbox-navigation-android/pull/5743. Changed name of the PR

LukasPaczos commented on libnavigation-base/src/main/java/com/mapbox/navigation/base/internal/route/NavigationRouteEx.kt:
```diff
@@ -45,16 +47,20 @@ fun NavigationRoute.updateLegAnnotations(
     val refreshedResponse = directionsResponse.toBuilder()
         .routes(refreshedRoutes)
         .build()
+    HttpServiceFactoryWrapper
```
can be removed?

RingerJK replied:
unnecessary changes, removed

LukasPaczos commented on libnavigation-base/src/main/java/com/mapbox/navigation/base/route/NavigationRoute.kt:
```diff
@@ -244,7 +307,32 @@ fun List<NavigationRoute>.toDirectionsRoutes() = map { it.directionsRoute }
+    ReplaceWith("toNavigationRoutes(routerOrigin)")
+)
+fun List<DirectionsRoute>.toNavigationRoutes() =
+    map { it.toNavigationRoute(RouterOrigin.Custom()) }
```
Wondering if we can be smarter about it. This is a compatibility function so I think we should make the best effort to figure out the source ourselves and only fallback to `Custom` when needed.

For example, if `route.uuid` is available, it's an `offboard` route.

RingerJK replied:
I've already sync with NN, and there's no good way to check what's route origin, it also might be Custom and we make change it with a false prediction

LukasPaczos replied:
That's where we have the other method that does accept origin as parameter.

This parameter-less function is here to provide our best compatibility with the old router interfaces (and we can document it this way), so I still think we should at least check the UUID.

RingerJK replied:
that's actually a good point of investigation: what if a user stored the route locally (as we do in mobile-metrics actually) and set it to the SDK with `uuid` and `RouterOrigin.OffBoard` NN will assume one as `Offboard` router with annotations and incidents and trying to update it but cannot.

RingerJK replied:
> This parameter-less function is here to provide our best compatibility with the old router interfaces (and we can document it this way), so I still think we should at least check the UUID.

We might assume the following cases:
- a route has UUID and it contains `@local` string -> route is onBoard
- a route has UUID -> offBoard
- else -> Custom

cc @LukasPaczos @mskurydin to check if it makes sense

mskurydin replied:
The UUID is not well documented and its contents is subject to change. Not sure we can rely on it for this.

LukasPaczos replied:
As discussed internally, let's merge as is. The only thing we'll need to remember is not make the origin a strong condition for other features :+1:

LukasPaczos commented on libnavigation-base/src/main/java/com/mapbox/navigation/base/route/NavigationRoute.kt:
```diff
@@ -260,11 +348,37 @@ fun List<DirectionsRoute>.toNavigationRoutes() = map { it.toNavigationRoute() }
 fun DirectionsRoute.toNavigationRoute(): NavigationRoute = this.toNavigationRoute(
-    NativeRouteParserWrapper
+    NativeRouteParserWrapper,
+    RouterOrigin.Custom(),
```
Same as above.

LukasPaczos commented on CHANGELOG.md:
```diff
@@ -7,6 +7,12 @@ Mapbox welcomes participation and contributions from everyone.
+  - method `NavigationRoute#create(String, String)` by `NavigationRoute#create(String, String, RouterOrigin)`;
+  - extension `List<DirectionsRoute>#toNavigationRoutes()` by `List<DirectionsRoute>#toNavigationRoutes(RouterOrigin)`;
+  - extension `DirectionsRoute#toNavigationRoute()` by `DirectionsRoute#toNavigationRoute(RouterOrigin)`.
+- Fixed runtime exception on [replacing the default logger module](https://docs.mapbox.com/android/navigation/guides/get-started/modularization/#logger) during library loading. [#5738](https://github.com/mapbox/mapbox-navigation-android/pull/5738)
```
```suggestion
- Fixed an issue where [replacing the default logger module](https://docs.mapbox.com/android/navigation/guides/get-started/modularization/#logger) was throwing a runtime exception during library loading. [#5738](https://github.com/mapbox/mapbox-navigation-android/pull/5738)
```
